src/nunit-gui/Presenters/TestPropertiesPresenter.cs

```diff
@@ -105,27 +106,43 @@ private void DisplayTestProperties()
                     _view.ElapsedTime = resultNode.Duration.ToString("f3");
                     _view.AssertCount = resultNode.AssertCount.ToString();
 
-                    var assertions = resultNode.Xml.SelectNodes("assertions/assertion");
-                    XmlNode message = null;
-                    XmlNode stackTrace = null;
+                    var assertionNodes = resultNode.Xml.SelectNodes("assertions/assertion");
```

Comment:
The nesting depth and changes I feel would be better understood with a few more private methods to help break up this complex area